test/spec/features/modeling/CreateConnectionSpec.js

```diff
@@ -118,6 +118,89 @@ describe('features/modeling - create connection', function() {
     });
 
 
+    describe('should create (connection -> shape)', function() {
+
+      var sourceConnection;
+
+      beforeEach(inject(function(elementFactory, canvas, modeling) {
+        var helperSource = elementFactory.createShape({
+          id: 'helper1',
+          x: 310, y: 110, width: 40, height: 40
+        });
+
+        var helperTarget = elementFactory.createShape({
+          id: 'helper2',
+          x: 310, y: 210, width: 40, height: 40
+        });
+
+        canvas.addShape(helperSource, parentShape);
+        canvas.addShape(helperTarget, parentShape);
+
+        sourceConnection = elementFactory.createConnection({
+          id: 'sourceConnection'
+        });
+
+        modeling.connect(helperSource, helperTarget, sourceConnection);
+      }));
+
+      it('execute', inject(function(modeling, elementRegistry) {
+
+        // when
+        var createdConnection = modeling.connect(sourceConnection, targetShape, newConnection);
+
+        var connection = elementRegistry.get('newConnection');
+
+        // then
+        expect(createdConnection).to.eql(connection);
+
+        expect(connection.id).to.eql('newConnection');
+
+        expect(connection.waypoints).to.eql([
```

Comment:
Should we also verify `connection.{incoming,outgoing}`

Reply:
Makes sense! cf. [ab73076](https://github.com/bpmn-io/diagram-js/pull/641/commits/ab73076f5aef37d549fc6a922a4b494ad7be42e1)